components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -335,6 +339,11 @@ class GeckoEngineSession(
         override fun onCloseRequest(session: GeckoSession) = Unit
 
         override fun onTitleChange(session: GeckoSession, title: String) {
+            // Only track history meta information if we're not in a private mode.
+            if (!privateMode) {
+                settings.historyTrackingDelegate?.onTitleChanged(currentUrl, title)
+            }
```

Comment:
Ignoring private pages is something the storage wants. Is it correct to assume that every `HistoryTrackingDelegate` implementation will not care about them?

Reply:
Probably not! For general browsing scenarios this is exactly what we want though, I think?

We could expand this to include “isPrivate” flag, and have the delegate implementation do what it wants.

What do you think?

Reply:
Changed the delegate interface to require passing `privateMode`. This gives us the most flexibility around different types of delegate implementations, and keeps the engine stuff simpler.